Src/DependencyCollector/Shared/Implementation/DependencyTableStore.cs

```diff
@@ -29,8 +31,18 @@ internal static DependencyTableStore Instance
         {
            get 
            {
-              Interlocked.CompareExchange<DependencyTableStore>(ref instance, new DependencyTableStore(), null);
-              return instance;
+               if (instance == null)
+               {
+                   lock (SyncRoot)
+                   {
+                       if (instance == null)
```

Comment:
Double checking is described in that article. Anyway I'm fine with @cijothomas PR #559. We should have it in the stable release